aws-codeguruprofiler-profilinggroup/src/main/java/software/amazon/codeguruprofiler/profilinggroup/CreateHandler.java

```diff
@@ -146,4 +156,34 @@ private static void safelyInvokeApi(final Runnable lambda) {
         }
         return Optional.of(model.getAgentPermissions().getPrincipals());
     }
+
+    private static CreateProfilingGroupRequest getCreateProfilingGroupRequest(
+        final String pgName,
+        final String computePlatform,
+        final String requestToken,
+        final Optional<Map<String, String>> tags
+    ) {
+        if (tags.isPresent()) {
+            return CreateProfilingGroupRequest.builder()
+                       .profilingGroupName(pgName)
+                       .computePlatform(computePlatform)
+                       .clientToken(requestToken)
+                       .tags(tags.get())
+                       .build();
+        }
+        return CreateProfilingGroupRequest.builder()
+                   .profilingGroupName(pgName)
+                   .computePlatform(computePlatform)
+                   .clientToken(requestToken)
+                   .build();
+    }
+
+    private static Optional<Map<String, String>> tagsFromModel(final ResourceModel model) {
+        List<Tag> tags = model.getTags();
+        if (tags == null || tags.isEmpty()) {
+            return Optional.empty();
+        }
+
+        return Optional.of(tags.stream().collect(Collectors.toMap(Tag::getKey, Tag::getValue)));
+    }
```

Comment:
Since no tags is the same as empty tags, I suggest not using an optional and returning always a map. When there are no tags -- just return a n empty map.